boards/nano33ble-bootloader/src/main.rs

```diff
@@ -117,7 +117,7 @@ pub unsafe fn reset_handler() {
 
     let active_notifier_led = static_init!(
         kernel::hil::led::LedHigh<'static, nrf52840::gpio::GPIOPin>,
-        kernel::hil::led::LedHigh::new(&base_peripherals.gpio_port[LED_ON_PIN])
+        kernel::hil::led::LedHigh::new(&nrf52840_peripherals.gpio_port[LED_ON_PIN])
```

Comment:
I does not compile otherwise, am I missing something?

Reply:
> Hmm, ok, so it seems that `abort()` is being called too early? The easy test would be to increase the timeout on line 208. That slows down the bootloader, but shouldn't affect its functionality.

It is called ok, just that when it is called, it seems that no data is being passed.

Reply:
I'm trying to understand where the 44 byte limit is coming from. To get 44 bytes, I would expect two uart.receive_buffer calls to complete:

1. The first for 1 byte.
2. The second for 43 bytes.

However, since the second asked for 256 bytes but only received 43, an `abort()` must have been called. If the abort is called too early then that would explain why the bootloader layer only receives 44 bytes when more were sent by the host.

Using the nrf52840dk I tried various receive and abort combinations, with various buffer sizes in the kernel, and everything worked as expected. So it seems like the `nrf52/src/uart.rs` implementation for receive and abort is working. So what I still can't understand is why you are only seeing up to 44 bytes received before receive_multiple passes the buffer up to the bootloader layer.

Reply:
> I'm trying to understand where the 44 byte limit is coming from. To get 44 bytes, I would expect two uart.receive_buffer calls to complete:
> 
>     1. The first for 1 byte.
> 
>     2. The second for 43 bytes.
> 
> 
> However, since the second asked for 256 bytes but only received 43, an `abort()` must have been called. If the abort is called too early then that would explain why the bootloader layer only receives 44 bytes when more were sent by the host.
> 
> Using the nrf52840dk I tried various receive and abort combinations, with various buffer sizes in the kernel, and everything worked as expected. So it seems like the `nrf52/src/uart.rs` implementation for receive and abort is working. So what I still can't understand is why you are only seeing up to 44 bytes received before receive_multiple passes the buffer up to the bootloader layer.

44 bytes is the number if bytes that it can read before the timer expires. The microbit has an nrf52833 mcu.